pkg/provider/utils.go

```diff
@@ -3,7 +3,10 @@ package provider
 import (
 	"errors"
 
+	"fmt"
+
 	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/api"
+	diag2 "github.com/hashicorp/terraform-plugin-framework/diag"
```

Comment:
```suggestion
	frameworkdiag "github.com/hashicorp/terraform-plugin-framework/diag"
```
Instead of diag2, let's be more specific on where this diag comes from, as we did in provider.go.
Please update the fromErr function accordingly.